src/Query.ts

```diff
@@ -202,8 +202,17 @@ export function map<U, L, A, B>(
   fa: ObservableQuery<U, L, A>,
   f: (a: A) => B
 ): ObservableQuery<U, L, B> {
-  return compose(
-    fa,
-    queryStrict(a => taskEither.of<L, B>(f(a)), available)
-  );
+  return queryStrict(a => fa.run(a).map(f), available);
```

Comment:
seeing this I feel the urge to change to name of `compose` to `chain` :D

Reply:
😂 yes I know, even though this is not 100% correct in fp-ts lingo. See my "side note" here: https://github.com/buildo/avenger/issues/201#issuecomment-524835612